### PR TITLE
samples: nrf9160: Add single-cell (SCELL) positioning to AGPS

### PR DESCRIPTION
**Pull request by junqingzou**
Defined two flags in project config:
.CONFIG_NRF_CLOUD_AGPS_REQ_CELL_BASED_LOC=y
.CONFIG_AGPS_SINGLE_CELL_ONLY=y
Parse the returned coordinates from nRF Cloud.

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

**Comment by tokangas** [comment]
Cellular positioning doesn't belong to the AGPS sample or AGPS library. Cellular positioning is not related to (A)GPS, but an alternative.

In my opinion, we should not add this to the AGPS sample and also remove the cellular positioning from the AGPS library. There's support also in nRFCloud library under AGPS, which is not the correct place for it. I think a new library should be created for cellular positioning.

**Comment by junqingzou** [comment]
> Cellular positioning doesn't belong to the AGPS sample or AGPS library. Cellular positioning is not related to (A)GPS, but an alternative.
> 
> In my opinion, we should not add this to the AGPS sample and also remove the cellular positioning from the AGPS library. There's support also in nRFCloud library under AGPS, which is not the correct place for it. I think a new library should be created for cellular positioning.

The call to get SCELL location could be made from application to nrf_cloud directly
#if defined(CONFIG_AGPS_SINGLE_CELL_ONLY)
	err = nrf_cloud_agps_request_cell_location(CELL_LOC_TYPE_SINGLE,
		(bool)IS_ENABLED(CONFIG_NRF_CLOUD_AGPS_REQ_CELL_BASED_LOC));

And the return in CLOUD_EVT_DATA_RECEIVED could be parsed by calling nrf_cloud_agps_process(), and final result be fetched by nrf_cloud_agps_get_last_cell_location(). That is to say, SCELL positioning does not really need the help of ncs\lib\agps. NOTE you have noticed the name of nrf_cloud APIs, always with _agps_ even for cellular positioning.

**Comment by junqingzou** [comment]
I created another sample project master\nrf\samples\nrf9160\scell
.Direct API call to nrf_cloud_agps
.Removed dependency on lib\agps and drivers\gps\nrf9160_gps
.Project config
  CONFIG_NRF_CLOUD_AGPS=y
  CONFIG_NRF_CLOUD_AGPS_SINGLE_CELL_ONLY=y
  CONFIG_NRF_CLOUD_AGPS_REQ_CELL_BASED_LOC=y
.Send SCELL request after  CLOUD_EVT_READY
  // by a system work, which require large stack due to modem_info used in nrf_cloud API

Test result:
I: CLOUD_EVT_READY
I: Service info sent to cloud
D: Created A-GPS request: {"appId":"SCELL","messageType":"DATA","data":{"mcc":440,"mnc":20,"tac":4202,"eci":10720861,"phycid":0}}
D: A-GPS request sent
I: CLOUD_EVT_DATA_RECEIVED
D: Cell location: (35.455833, 139.626111), unc: 1582, type: 0
I:
Longitude:  35.455833
Latitude:   139.62611

**Comment by tokangas** [comment]
Maybe we should create a new sample for cellular positioning. I think the nRFCloud library needs some refactoring because cellular positioning is now under AGPS, but we can update the sample when that's done.

File: samples/nrf9160/agps/src/main.c [hunk 1]
```diff
@@ -146,15 +148,29 @@ static void cloud_event_handler(const struct cloud_backend *const backend,
...
+			double lat, lon;
+
+			err = gps_get_last_cell_location(&lat, &lon);
+			if (err) {
+				LOG_ERR("Failed to get GPS data, error: %d", err);
+				break;
+			}
+
+			LOG_INF("\nLongitude:  %f\nLatitude:   %f\n", lat, lon);
+			gps_stop(gps_dev);
```
Comment (jayteemo) [inline samples/nrf9160/agps/src/main.c:171]: if `CONFIG_AGPS_SINGLE_CELL_ONLY` is enabled there is no need to call `gps_start()`

Comment (junqingzou) [reply samples/nrf9160/agps/src/main.c:171]: In this sample, gps_agps_request() is called from gps_handler(GPS_EVT_AGPS_DATA_NEEDED), so we still need to start GPS.
Sequence in the log:
I: CLOUD_EVT_READY
I: Service info sent to cloud
D: GPS mode is enabled
D: GPS operational
I: Periodic GPS search started with interval 240 s, timeout 120 s
I: GPS_EVT_SEARCH_STARTED
D: A-GPS data update needed
I: GPS_EVT_AGPS_DATA_NEEDED
D: Created A-GPS request: {"appId":"SCELL","messageType":"DATA","data":{"mcc":440,"mnc":20,"tac":4202,"eci":10720861,"phycid":0}}
D: A-GPS request sent

But SCELL does not really need GPS so in the sample project I made above, no GPS started.

**Comment by junqingzou** [comment]
Close this PR as a total solution is handled in PR4340 instead.

**Pull request closed without merging**